### ISSUE TITLE
why use the sigmoid？

### ISSUE THREAD

**Issue by bringtree** [carpedm20/DCGAN-tensorflow#222]
Hello man, 
in model.py
line 111      `self.D_, self.D_logits_ = self.discriminator(self.G, self.y, reuse=True)`
line 311   
<img width="624" alt="pastedgraphic-3" src="https://user-images.githubusercontent.com/20106478/30519019-18973842-9bbf-11e7-918c-2eb6f5e176d6.png">
d_loss 
<img width="569" alt="pastedgraphic-2" src="https://user-images.githubusercontent.com/20106478/30519022-22b9a65c-9bbf-11e7-84bf-cac3b38efc50.png">
is it a bug ?
may be it should be 
      sigmoid_cross_entropy_with_logits(self.D_logits, tf.ones_like(self.D_logits)))
and we need not to calculate sigmoid(h4).

i am a noob .  if i am wrong, i hope that you get tell me why ？
thanks.

**Comment by Feiyu-Zhang** [comment]
> Hello man,
> in model.py
> line 111 `self.D_, self.D_logits_ = self.discriminator(self.G, self.y, reuse=True)`
> line 311
> <img alt="pastedgraphic-3" width="624" src="https://user-images.githubusercontent.com/20106478/30519019-18973842-9bbf-11e7-918c-2eb6f5e176d6.png">
> d_loss
> <img alt="pastedgraphic-2" width="569" src="https://user-images.githubusercontent.com/20106478/30519022-22b9a65c-9bbf-11e7-84bf-cac3b38efc50.png">
> is it a bug ?
> may be it should be
> sigmoid_cross_entropy_with_logits(self.D_logits, tf.ones_like(self.D_logits)))
> and we need not to calculate sigmoid(h4).
> 
> i am a noob . if i am wrong, i hope that you get tell me why ？
> thanks.

Discriminator outputs probability by using calculating sigmoid(h4).
